### DISCUSSION TITLE
Handling the size of the generated schemas

### DISCUSSION THREAD

**Discussion by joacoespinosa** [chrishoermann/zod-prisma-types#190] [Q&A] [answered]
This library seems to offer everything I need to handle validation based on models generated from my database definition.

However, I get 763 files (I'm currently trying `useMultipleFiles=true`) and VS Code simply can't handle it. The compiler won't highlight errors and every time I hover on a specific type, I get a everlasting `Loading...` – not to mention that navigating to said type is simply unavailable –

Is there any advice/guidance you can provide regarding how to efficiently utilize the produced types/schemas?

My database schema only has eight tables, although next-auth uses four additional ones. Just in case it is useful, I'm also using `createRelationValuesTypes=true` and `createInputTypes=true`

And sorry if this comment is funny but I'm utilizing a decent enough computer.

Thank you in advance

**Comment by chrishoermann** [comment, marked as answer]
@joacoespinosa no worries, this comment is not funny. I sometimes experience this behaviour myself and it can be annoying. I sadly do not have a bulletproof method how this behavior of typescript could be improved since there is a lot of type inference going on with zod in general that can lead to a degrading typescript perfomance - the amount of generated schemas does not help with this.

Sadly there is no solution to reduce the amount of schemas other than disabling input schemas in general with `createInputTypes = false` and create necessary validation schemas e.g. for `trpc` or `react-hook-forms` by hand. 

The problem lies in the complexity of the data in the `prisma dmmf` which is the basis for all `prisma types` and which is also used to generate the corresponding input schemas.  Since relational date can be heavily nested and prisma provides extensive filter capabilities it is also not possible to effectively omit some types without introducing bugs, type errors and unwanted behaviour. 

So sadly you'll have to live with this problem for now. 😞

**Reply by joacoespinosa** [reply]
Thank you, @chrishoermann. I was able to confirm, unfortunately, that using deep/complex types is just not possible when managing forms through `react-hook-form`. Typescript simply can't handle it.

In case this comment is of help to anyone experiencing the same issue, I ended up using some of the "plainer" types (e.g., `EntityCreateInputSchema`, which also contains relation value types useful for initializing nested relations) as base validators, and adding customizations by hand as you said. I am, however, able to use the deep/complex types on the backend for validating requests that do _not_ originate off of web forms, utilizing `zod-prisma-types` as a sort of API contract if you will.

**Comment by julio-ccatb** [comment]
I encountered a similar problem, but I discovered that you can utilize either the Model Schema or the type itself for the form values. Simply employ the CreateInputSchema with the Zod validator, assuming that's what you're using.

Try something like this 

```Javascript
  type Input = z.infer<typeof TbaseSchema>;
  const resolver = zodResolver(TbaseCreateWithoutBaseInputSchema);

  const {
    register,
    handleSubmit,
    reset,
    clearErrors,
    formState: { errors },
  } = useForm<Input>({ resolver });
```
